### PR TITLE
Add examples/hosted-wallet-kt

### PR DESCRIPTION
**Pull request by KendallWeihe**
Please refer to the `examples/hosted-wallet-kt/README.md` for context.

> [!WARNING]
> 
> The README includes steps to use the `gradle` command line tool, but this is known to not function at this time. Instead, for current usage, use IntelliJ. Build systems is the next priority on my list.

---

Also includes changes for:

- `Map<String, Any>` instead of `JsonNode` (this way we don't leak abstractions from transitive dependencies)
- Binding for `CreateExchangeRequestBody` 🚧 We'll need to add this type to the APID at a later date (reconciliation is on my list of items)
- Latest `web5` includes `PortableDid` features
- Rename polynorphic conversion struct's to `ToOuter*`
- Bug fix in `crates/tbdex/src/messages/rfq.rs` with regards to Offering verification

### CODE REVIEW

**Comment by KendallWeihe** [comment]
Subsequently, I'd also like to add functionality to the `web5` CLI to better enable adminstrative matters (specifically matters of DID Service's and VC types). I'd also like to ideally automate the `mvn install` step.

File: bound/kt/src/main/kotlin/tbdex/sdk/messages/Rfq.kt [hunk 1]
```diff
@@ -80,13 +79,13 @@ data class CreateRfqData(
 
 data class CreateSelectedPayinMethod(
     val kind: String,
-    val paymentDetails: JsonNode? = null,
+    val paymentDetails: Map<String, Any?>? = null,
```
Comment (nitro-neal) [inline bound/kt/src/main/kotlin/tbdex/sdk/messages/Rfq.kt:82]: yup good call axing JsonNode

File: bound/kt/src/test/kotlin/tbdex/sdk/httpclient/E2ePfiExemplarIntegrationTest.kt [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-import org.junit.jupiter.api.Assertions.fail
+import org.junit.jupiter.api.Disabled
 import org.junit.jupiter.api.Test
 import tbdex.sdk.messages.*
-import tbdex.sdk.rust.RustCoreException
-import tbdex.sdk.rust.Web5RustCoreException
 import tbdex.sdk.web5.*
 
 class E2ePfiExemplarIntegrationTest {
+    @Disabled
```
Comment (nitro-neal) [inline bound/kt/src/test/kotlin/tbdex/sdk/httpclient/E2ePfiExemplarIntegrationTest.kt:10]: could add a line of documentation explaining that you have to spin up a tbdex-exemplar and then run this test

File: examples/hosted-wallet-kt/pfi/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+repositories {
+    mavenCentral()
+    mavenLocal()
+}
+
+dependencies {
+    implementation(kotlin("stdlib"))
+    implementation("com.sparkjava:spark-core:2.9.4")
+    implementation("com.squareup.okhttp3:okhttp:4.12.0")
+    implementation("tbdex.sdk.core:tbdex-core-kt:1.0-SNAPSHOT")
```
Comment (nitro-neal) [inline examples/hosted-wallet-kt/pfi/build.gradle.kts:15]: ✅

File: examples/hosted-wallet-kt/pfi/src/main/kotlin/api/Exchanges.kt [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        val rfq = requestBody.rfq
+
+        rfq.verifyOfferingRequirements(this.offeringsRepository.getOffering(rfq.data.offeringId))
+
+        this.exchangesToReplyTo[rfq.metadata.exchangeId] = replyTo
+
+        res.status(202)
+
+        Thread {
+            Thread.sleep(3000)
```
Comment (nitro-neal) [inline examples/hosted-wallet-kt/pfi/src/main/kotlin/api/Exchanges.kt:43]: should there be retries?

File: examples/hosted-wallet-kt/pfi/src/main/kotlin/data/Offerings.kt [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import tbdex.sdk.web5.InputDescriptor
+import tbdex.sdk.web5.Constraints
+import tbdex.sdk.web5.Field
+import tbdex.sdk.web5.Filter
+
+class Offerings(private val bearerDid: BearerDid) {
+    private var offerings: MutableList<Offering> = mutableListOf()
+
+    init {
+        // this app currently always seeds the fake data
```
Comment (nitro-neal) [inline examples/hosted-wallet-kt/pfi/src/main/kotlin/data/Offerings.kt:20]: i think eventually we should put these hardcoded things in a resource folder as a .json, will be easier to see the full picture / edit / swap out / test

File: examples/hosted-wallet-kt/pfi/src/main/resources/.env [hunk 6]
```diff
@@ -0,0 +1 @@
+PFI_PORTABLE_DID={"uri":"did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y","document":{"id":"did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y","verificationMethod":[{"id":"did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y#0","type":"JsonWebKey","controller":"did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y","publicKeyJwk":{"alg":"Ed25519","kty":"OKP","crv":"Ed25519","x":"BYEFUF5bxXrwwuKUiyamOwYck_pDj4C3OOo_nYNWyzQ"}}],"authentication":["did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y#0"],"assertionMethod":["did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y#0"],"capabilityInvocation":["did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y#0"],"capabilityDelegation":["did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y#0"],"service":[{"id":"did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y#pfi","type":"PFI","serviceEndpoint":["http://localhost:8082"]}]},"privateKeys":[{"alg":"Ed25519","kty":"OKP","crv":"Ed25519","d":"EBvZ0Uw1MpddCT5Nqiyu1KBOAYQXvQxkXmInSVkAkzc","x":"BYEFUF5bxXrwwuKUiyamOwYck_pDj4C3OOo_nYNWyzQ"}]}
```
Comment (nitro-neal) [inline examples/hosted-wallet-kt/pfi/src/main/resources/.env:1]: +1 much better than dynamically created every time lol, just one and done

File: examples/hosted-wallet-kt/wallet/src/main/resources/.env [hunk 7]
```diff
@@ -0,0 +1,3 @@
+PFI_DID_URI=did:dht:ysyokwn6mxnzihgnhkkesjig8cdb3r94eq8abp3a7e935y4s3c4y
+VERIFIABLE_CREDENTIAL=eyJ0eXAiOiJKV1QiLCJhbGciOiJFZERTQSIsImtpZCI6ImRpZDpkaHQ6eXN5b2t3bjZteG56aWhnbmhra2VzamlnOGNkYjNyOTRlcThhYnAzYTdlOTM1eTRzM2M0eSMwIn0.eyJ2YyI6eyJAY29udGV4dCI6WyJodHRwczovL3d3dy53My5vcmcvMjAxOC9jcmVkZW50aWFscy92MSJdLCJpZCI6InVybjp2Yzp1dWlkOjRiOTUwOWE4LTQwMjgtNDRkOC05OGE0LWRiODg1MjhmNjY4YyIsInR5cGUiOlsiVmVyaWZpYWJsZUNyZWRlbnRpYWwiLCJTYW5jdGlvbkNyZWRlbnRpYWwiXSwiaXNzdWVyIjoiZGlkOmRodDp5c3lva3duNm14bnppaGduaGtrZXNqaWc4Y2RiM3I5NGVxOGFicDNhN2U5MzV5NHMzYzR5IiwiaXNzdWFuY2VEYXRlIjoiMjAyNC0wNy0wMlQwNDoyNDoxNC4yNzYzMjUrMDA6MDAiLCJleHBpcmF0aW9uRGF0ZSI6bnVsbCwiY3JlZGVudGlhbFN1YmplY3QiOnsiaWQiOiJkaWQ6ZGh0OjFmczVobnhzZ3R4Z2RyNHd6cWkzOGNuajQ2YjF3aGhuOTRvandvNjZnOGhzYzVidDNmZ3kifX0sImlzcyI6ImRpZDpkaHQ6eXN5b2t3bjZteG56aWhnbmhra2VzamlnOGNkYjNyOTRlcThhYnAzYTdlOTM1eTRzM2M0eSIsImp0aSI6InVybjp2Yzp1dWlkOjRiOTUwOWE4LTQwMjgtNDRkOC05OGE0LWRiODg1MjhmNjY4YyIsInN1YiI6ImRpZDpkaHQ6MWZzNWhueHNndHhnZHI0d3pxaTM4Y25qNDZiMXdoaG45NG9qd282Nmc4aHNjNWJ0M2ZneSIsIm5iZiI6MTcxOTg5NDI1NCwiaWF0IjoxNzE5ODk0MjU0fQ.c4ws9jR28jElo_uaW9l5OTL-IPMx4JxWl4De7l_BTk0qNhcFlRtR-U0b9087CUOdpNu25XGZzn-R_EVImRGgCw
```
Comment (nitro-neal) [inline examples/hosted-wallet-kt/wallet/src/main/resources/.env:2]:
```suggestion
HOSTED_WALLET_VERIFIABLE_CREDENTIAL=eyJ0eXAiOiJKV1QiLCJhbGciOiJFZERTQSIsImtpZCI6ImRpZDpkaHQ6eXN5b2t3bjZteG56aWhnbmhra2VzamlnOGNkYjNyOTRlcThhYnAzYTdlOTM1eTRzM2M0eSMwIn0.eyJ2YyI6eyJAY29udGV4dCI6WyJodHRwczovL3d3dy53My5vcmcvMjAxOC9jcmVkZW50aWFscy92MSJdLCJpZCI6InVybjp2Yzp1dWlkOjRiOTUwOWE4LTQwMjgtNDRkOC05OGE0LWRiODg1MjhmNjY4YyIsInR5cGUiOlsiVmVyaWZpYWJsZUNyZWRlbnRpYWwiLCJTYW5jdGlvbkNyZWRlbnRpYWwiXSwiaXNzdWVyIjoiZGlkOmRodDp5c3lva3duNm14bnppaGduaGtrZXNqaWc4Y2RiM3I5NGVxOGFicDNhN2U5MzV5NHMzYzR5IiwiaXNzdWFuY2VEYXRlIjoiMjAyNC0wNy0wMlQwNDoyNDoxNC4yNzYzMjUrMDA6MDAiLCJleHBpcmF0aW9uRGF0ZSI6bnVsbCwiY3JlZGVudGlhbFN1YmplY3QiOnsiaWQiOiJkaWQ6ZGh0OjFmczVobnhzZ3R4Z2RyNHd6cWkzOGNuajQ2YjF3aGhuOTRvandvNjZnOGhzYzVidDNmZ3kifX0sImlzcyI6ImRpZDpkaHQ6eXN5b2t3bjZteG56aWhnbmhra2VzamlnOGNkYjNyOTRlcThhYnAzYTdlOTM1eTRzM2M0eSIsImp0aSI6InVybjp2Yzp1dWlkOjRiOTUwOWE4LTQwMjgtNDRkOC05OGE0LWRiODg1MjhmNjY4YyIsInN1YiI6ImRpZDpkaHQ6MWZzNWhueHNndHhnZHI0d3pxaTM4Y25qNDZiMXdoaG45NG9qd282Nmc4aHNjNWJ0M2ZneSIsIm5iZiI6MTcxOTg5NDI1NCwiaWF0IjoxNzE5ODk0MjU0fQ.c4ws9jR28jElo_uaW9l5OTL-IPMx4JxWl4De7l_BTk0qNhcFlRtR-U0b9087CUOdpNu25XGZzn-R_EVImRGgCw
```

File: examples/hosted-wallet-kt/wallet/src/main/resources/.env [hunk 8]
```diff
@@ -0,0 +1,3 @@
...
+PORTABLE_DID_JSON={"uri":"did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy","document":{"id":"did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy","verificationMethod":[{"id":"did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0","type":"JsonWebKey","controller":"did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy","publicKeyJwk":{"crv":"Ed25519","kty":"OKP","x":"kW2-CfY0XmGTVLurk7BJ14Mqc4L-oJpD3jH5ZmwxyUw","kid":"ezoEr4cxqaYa9eOA3YkvCL1kw9yUuCYl3KMKO79sIbI","alg":"EdDSA"}}],"authentication":["did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0"],"assertionMethod":["did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0"],"capabilityDelegation":["did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0"],"capabilityInvocation":["did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0"]},"privateKeys":[{"crv":"Ed25519","d":"jVOdpSIN-DhddW_XVnDipukuzu6-8zieXQtkECZYJ04","kty":"OKP","x":"kW2-CfY0XmGTVLurk7BJ14Mqc4L-oJpD3jH5ZmwxyUw","kid":"ezoEr4cxqaYa9eOA3YkvCL1kw9yUuCYl3KMKO79sIbI","alg":"EdDSA"}]}
```
Comment (nitro-neal) [inline examples/hosted-wallet-kt/wallet/src/main/resources/.env:3]:
```suggestion
HOSTED_WALLET_PORTABLE_DID_JSON={"uri":"did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy","document":{"id":"did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy","verificationMethod":[{"id":"did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0","type":"JsonWebKey","controller":"did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy","publicKeyJwk":{"crv":"Ed25519","kty":"OKP","x":"kW2-CfY0XmGTVLurk7BJ14Mqc4L-oJpD3jH5ZmwxyUw","kid":"ezoEr4cxqaYa9eOA3YkvCL1kw9yUuCYl3KMKO79sIbI","alg":"EdDSA"}}],"authentication":["did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0"],"assertionMethod":["did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0"],"capabilityDelegation":["did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0"],"capabilityInvocation":["did:dht:1fs5hnxsgtxgdr4wzqi38cnj46b1whhn94ojwo66g8hsc5bt3fgy#0"]},"privateKeys":[{"crv":"Ed25519","d":"jVOdpSIN-DhddW_XVnDipukuzu6-8zieXQtkECZYJ04","kty":"OKP","x":"kW2-CfY0XmGTVLurk7BJ14Mqc4L-oJpD3jH5ZmwxyUw","kid":"ezoEr4cxqaYa9eOA3YkvCL1kw9yUuCYl3KMKO79sIbI","alg":"EdDSA"}]}
```

**Review by nitro-neal — APPROVED** [review]
Looks great! Im pulling down on my side and going to test locally

File: examples/hosted-wallet-kt/wallet/src/main/kotlin/Webhook.kt [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+    init {
+        port(8081)
+
+        post("/pfi-reply-to") { req, res ->
+            // todo we need parse support
+
+            val requestBodyString = req.body()
+
+            try {
```
Comment (KendallWeihe) [inline examples/hosted-wallet-kt/wallet/src/main/kotlin/Webhook.kt:21]: this is not intended to be the finished product

File: examples/hosted-wallet-kt/pfi/src/main/kotlin/api/Exchanges.kt (hunk 4 above)
Comment (KendallWeihe) [reply examples/hosted-wallet-kt/pfi/src/main/kotlin/api/Exchanges.kt:43]: maybe eventually, but it's. a bit much right now. prioritizing "happy path" to maximize ease of learning

File: examples/hosted-wallet-kt/pfi/src/main/kotlin/data/Offerings.kt (hunk 5 above)
Comment (KendallWeihe) [reply examples/hosted-wallet-kt/pfi/src/main/kotlin/data/Offerings.kt:20]: yeah I like this idea a lot, if you want to take the reigns then please do

**Pull request merged**